test: Integration tests for v2

stanleyjones opened the pull request:
See #103.

This PR includes the following tests, running against the latest `many-rs` and all the changes required to get them to pass (which is mostly reverting #99 and replacing every `Uint8Array` with `Buffer` again).

- Can we send an already encoded request and get a response?
- Can we encode a request from bytes and get a response?
- Can we form a request from an object and get a response?
- Can we call a method to form a request and get a response?
- Can we use a shortcut method to form a request and get a response?
- Can we sign a request and get a response (the rest are all anonymous)?

### Test Results

(Currently skipping the KeyValue tests and concentrating on the above.)

```
❯ npm run test:integration

> @liftedinit/many-js@0.1.0-alpha.9 test:integration
> jest --testMatch '**/?(*.)+(integration).[jt]s?(x)'

 PASS  src/__tests__/base-service.integration.ts

Test Suites: 1 skipped, 1 passed, 1 of 2 total
Tests:       3 skipped, 6 passed, 9 total
Snapshots:   0 total
Time:        0.967 s, estimated 1 s
Ran all test suites.
```

fmorency commented on src/__tests__/data.ts:
```diff
@@ -0,0 +1,12 @@
+import { KeyPair } from "../id";
+
+export const SERVERS = {
+  LEDGER: "http://127.0.0.1:8001",
```
You should point ports to the ABCI

stanleyjones replied:
Done

fmorency reviewed (changes requested):
You didn't ask for a review, but I'm giving you one anyway! I didn't review the code, as I believe you will need to integrate changes from #103 here.

For the sake of posterity, can you document why you needed to revert https://github.com/liftedinit/many-js/pull/99?

stanleyjones commented:
For the record: The CBOR library we're using assumes Buffers, so replacing them with Uint8Arrays meant that things were getting coded as such in CBOR (and tagged as TypedArrays). There are other CBOR libraries that don't make this assumption (like [cbor-x](https://www.npmjs.com/package/cbor-x)) that we could switch to in the future, but I didn't want to do that at the same time.

fmorency commented on src/__tests__/data.ts:
```diff
@@ -0,0 +1,12 @@
+
+export const SERVERS = {
+  LEDGER: "http://127.0.0.1:8000",
+  KEYVALUE: "http://127.0.0.1:8010",
```
The correct port for kvstore ABCI when running the `run.sh` script is `8011`.

fmorency commented on src/__tests__/keyvalue.integration.ts:
```diff
@@ -0,0 +1,57 @@
+import { KeyPair, KeyValueService, Request } from "..";
```
Is this still WIP? Should we remove this from the current PR and write a future kvstore-oriented PR?

fmorency commented on src/message/request/request.ts:
```diff
@@ -46,7 +47,7 @@ const requestMap: Transform = {
   5: ["timestamp", { fn: (value: number) => new cbor.Tagged(1, value) }],
   6: "id",
-  7: ["nonce", { fn: (value: string) => cbor.encode(value) }],
+  // 7: ["nonce", { fn: (value: string) => cbor.encode(value) }],
```
Remove?

fmorency commented on src/message/request/request.ts:
```diff
@@ -67,8 +68,9 @@ export class Request extends Message {
     }
     const defaults = {
-      version: 1,
+      // version: 1,
```
Remove?

fmorency commented on src/message/request/request.ts:
```diff
@@ -67,8 +68,9 @@ export class Request extends Message {
-      version: 1,
+      // version: 1,
       timestamp: Math.floor(Date.now() / 1000),
+      // nonce: makeRandomBytes(16),
```
Remove?

fmorency reviewed (approved):
I didn't review the code in-depth. LGTM, assuming comments are addressed. Thanks!

stanleyjones commented:
Acknowledging that there are unaddressed comments here that I'll tackle in the next PR.

- KeyValue port and tests
- Uncomment or remove commented code
